compiler/AST/baseAST.cpp

```diff
@@ -552,6 +552,37 @@ void BaseAST::printDocsDescription(const char *doc, std::ostream *file, unsigned
   }
 }
 
+// This method is the same for several subclasses of BaseAST, so it is defined
+// here on BaseAST. 'doc' is not defined as a member of BaseAST, so it must be
+// taken as an argument here.
+//
+// Will print a deprecation warning in the documentation, if one is not already
+// present.  Checks for "deprecat" in the original documentation string if it
```

Comment:
If I'm understanding this right, a dev could do something silly and write 'deprecat' and this would fire?

Reply:
Yeah, though I think that's likely to be a case where they meant to say "deprecate".  The intention is to catch "deprecate", "deprecated" and "deprecation" without having to perform multiple checks - I don't think there's other words with that sequence in them that could trigger a false positive.  Grammatically, I think it'll mostly just be "deprecated" in practice, though